lib/rouge/lexers/epp.rb

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*- #
+
+module Rouge
+  module Lexers
+    class EPP < TemplateLexer
+      title "EPP"
+      desc "Embedded Puppet template files"
+
+      tag 'epp'
+
+      filenames '*.epp'
+
+      def initialize(opts={})
+        @puppet_lexer = Puppet.new(opts)
+
+        super(opts)
+      end
+
+      start do
+        parent.reset!
+        @puppet_lexer.reset!
+      end
+
+      open  = /<%%|<%=|<%#|(<%-|<%)(\s*?\|)?/
```

Comment:
So there doesn't need to be any space between the `|` and the opening tags? In other words, `<%|` is valid?

You also don't need to use the lazy version of `*` here. It won't make a difference either way.

Reply:
`<%|` is indeed valid EPP syntax.